shopfloor/services/zone_picking.py

```diff
@@ -535,15 +544,22 @@ def _scan_source_location(
             )
         else:
             # if no move line, narrow the list of move lines on the scanned location
-            response = self._list_move_lines(location)
-            message = self.msg_store.location_empty(location)
+            response = self._list_move_lines(
+                location, no_package=scan_location_or_pack_first
+            )
+            if not scan_location_or_pack_first:
+                message = self.msg_store.location_empty(location)
+            else:
+                message = self.msg_store.location_empty_scan_package(location)
         return response, message
 
     def _find_product_in_location(self, location, product_id):
         """Find the prooducts in stock in given location move line in the location."""
         domain = [("location_id", "=", location.id)]
         if product_id:
             domain.append(("product_id", "=", product_id))
+        if self.work.menu.scan_location_or_pack_first:
```

Comment:
I would add a comment for our future selves:
```suggestion
        # As we scanned a location, we want to list products that are not part of packages
        if self.work.menu.scan_location_or_pack_first:
```

Reply:
Added